src/main/java/nerdhub/textilelib/event/entity/player/PlayerJoinCallback.java

```diff
@@ -0,0 +1,21 @@
+package nerdhub.textilelib.event.entity.player;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public interface PlayerJoinCallback {
+
+    Event<PlayerJoinCallback> EVENT = EventFactory.createArrayBacked(PlayerJoinCallback.class, listeners -> (world, player, pos) -> {
+        for (PlayerJoinCallback callback : listeners) {
+            callback.onPlayerJoin(world, player, pos);
+        }
+    });
+
+    /**
+     * Fired when a player joins a world
+     */
+    void onPlayerJoin(World world, PlayerEntity player, BlockPos pos);
```

Comment:
1. why not make the types for this be `ServerWorld` and `ServerPlayerEntity` respectively?
2. I don't think that BlockPos parameter is needed here, since Enitites usually don't spawn in the middle of a block.